types/tx/middleware.go

```diff
@@ -21,12 +22,42 @@ type ResponseSimulateTx struct {
 	Result  *sdk.Result
 }
 
+// Response is the tx response type used in middlewares.
+type Response struct {
+	GasWanted uint64
+	GasUsed   uint64
+	// MsgResponses is an array containing each Msg service handler's response
+	// type, packed in an Any. This will get proto-serialized into the `Data` field
+	// in the ABCI Check/DeliverTx responses.
+	MsgResponses []*codectypes.Any
+	Log          string
+	Events       []abci.Event
+}
+
+// Request is the tx request type used in middlewares.
+type Request struct {
+	Tx      sdk.Tx
```

Comment:
Will this be replaced by the concrete type?

Reply:
Yes! https://github.com/cosmos/cosmos-sdk/issues/10347